src/sql/workbench/browser/modal/modal.ts

```diff
@@ -465,13 +465,14 @@ export abstract class Modal extends Disposable implements IThemable {
 		DOM.append(this.layoutService.container, this._bodyContainer!);
 		this.setInitialFocusedElement();
 
-		this.disposableStore.add(DOM.addDisposableListener(document, DOM.EventType.KEY_UP, (e: KeyboardEvent) => {
+		this.disposableStore.add(DOM.addDisposableListener(document, DOM.EventType.KEY_DOWN, (e: KeyboardEvent) => {
 			let context = this._modalShowingContext.get()!;
 			if (context[context.length - 1] === this._staticKey) {
 				let event = new StandardKeyboardEvent(e);
 				if (event.equals(KeyCode.Enter)) {
 					this.onAccept(event);
 				} else if (event.equals(KeyCode.Escape)) {
+					DOM.EventHelper.stop(e, true);
 					this.onClose(event);
 				} else if (event.equals(KeyMod.Shift | KeyCode.Tab)) {
```

Comment:
What about these other events? Do the events propagating up cause any issues - for example does pressing tab/enter in the callout dialog cause any issues with other stuff also getting the event?

Reply:
@Charles-Gagnon I looked back (11months) on this file prior to my changes for the callout dialog. This code has not changed much -- these other events didn't have any code added to stop bubbling or propagation.

Reply:
I'm saying make sure to test that now. Previously modal dialogs weren't going to be commonly launched from places like Notebooks so it's possible that because of that we might see some other side effects that weren't discovered previously.